src/client-lib/utils/rendezvous-socket.js

```diff
@@ -0,0 +1,43 @@
+import EventEmitter from 'eventemitter3';
+
+class RendezvousSocket extends EventEmitter {
```

Comment:
I'd prefer to use composition (e.g. attaching to `EventEmitter` to `this.events`) over inheritance here. Thoughts?